css/product-listing.css

```diff
@@ -0,0 +1,366 @@
+/* Main container */
+
+.listing {
+  margin-block-start: 5rem;
+}
+
+.listing__wrapper {
+  grid-template-columns: 1fr;
+  inline-size: min(90%, 1200px);
+  margin-inline: auto;
+  align-items: start;
+  gap: 1rem;
+  grid-row-gap: 1.5rem;
+  position: relative;
+}
+
+.listing__form {
+  display: grid;
+  gap: 1.5rem;
+  margin-block: 4rem;
+}
+
+.listing__filters {
+  position: fixed;
+  inset-block-end: 0;
+  inset-inline-start: 0;
+  z-index: 15;
+  overflow: hidden;
+  overflow-y: scroll;
+  block-size: 60vh;
+  background-color: var(--surface2);
+  inline-size: 100%;
+  padding-inline: 1rem;
+  opacity: 0;
+  pointer-events: none;
+  transition: opacity 250ms ease-in-out;
+}
+
+.listing__filter-item {
+  padding-block-start: 0;
+  padding-inline: 1.2rem;
+  padding-block: 1rem;
+  border-radius: 6px;
+  border: 1px solid transparent;
+}
+
+.listing__filter-item:focus-within {
+  border: 1px solid rgba(32, 32, 32, 0.295);
+}
+
+
+/* Input group styling */
+
+.filters__group + .filters__group {
+  margin-block-start: 0.8rem;
+}
+
+.filters__group {
+  display: flex;
+  align-items: center;
+  gap: 1.5ch;
+}
+
+.filters__label {
+  font-size: 0.9rem;
+  inline-size: 100%;
+}
+
+.filters__checkbox {
+  accent-color: var(--clr-secondary);
+}
+
+legend {
+  font-size: 1rem;
+  font-weight: bold;
+}
+
+.filters__select {
+  display: block;
+  padding-inline: 1rem;
+}
+
+optgroup {
+  margin-block-start: 1rem;
+}
+
+/* Resetting card size */
+
+.card {
+  inline-size: unset;
+}
+
+/* listing products */
+
+.listing__products {
+  gap: 2rem;
+  --grid-auto-item-size: 195px;
+}
+
+/* ----------- */
+/* Product card */
+/* ---------- */
+
+.card {
+  border: 0.5px solid rgba(58, 58, 58, 0.123);
+}
+
+img {
+  max-inline-size: 100%;
+  aspect-ratio: 1 / 1.3;
+  object-fit: cover;
+}
+
+.d-flex {
+  display: flex;
+  align-items: center;
+  justify-content: start;
+  gap: 5px;
+}
+
+.card__info {
+  padding: 0.5rem 0.6rem;
+  text-align: left;
+  color: var(--text1);
+}
+
+.card__desc {
+  font-size: 0.8rem;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+}
+
+.card__name {
+  font-size: 0.9rem;
+  font-weight: bold;
+}
+
+.card__discountedPrice {
+  font-weight: 700;
+  font-size: 0.9rem;
+}
+
+.card__realPrice {
+  font-size: 0.7rem;
+  text-decoration: line-through;
+}
+
+.card__discount {
+  font-size: 0.8rem;
+  font-weight: 700;
+  color: #734124;
```

Comment:
Try using variables for colors. It makes the job of maintaining and reviewing code so much easier :)